File: 3.c
#include<stdio.h>
#include<math.h>

int main()
{
	long int i, j, lpf, num = 600851475143, count = 1;

	for(i = 1; i < ceil(sqrt(num)); i++) if(num % i == 0)
	{
		for(j = 2, count = 0; j < i; j++) if(i % j == 0) count++;
		if(count == 0) lpf = j;
	}

	printf("The largest prime factor is %ld", lpf);
	return 0;
}

File: 3.py
import math

num = 600851475143
count = 1
j = 2

for i in range(1, math.ceil(math.sqrt(num))):
	if(num % i == 0): # Factor check
		count = 0
		for j in range(2, i+1): #Prime Number check
			if(i % j == 0):
				count += 1
		if(count == 1):
			lpf = j

print("Largest Prime factor of " + str(num) + " is " + str(lpf))
